File: sfml2-space-invaders-master/Bullet.cpp
#include "Bullet.h"

Bullet::Bullet()
{

}

Bullet& Bullet::operator=(const Bullet& other)
{
	reversed = other.reversed;
	tex2d = other.tex2d;
	spriteRect = other.spriteRect;
	pos = other.pos;
	return *this;
}

Bullet::~Bullet()
{
}

void Bullet::LoadResources()
{
	if (!reversed)
	{
		tex2d.Load("rd/bullet.png");
		spriteRect = sf::IntRect(0, 0, tex2d.sprite.getGlobalBounds().width, tex2d.sprite.getGlobalBounds().height);
	}
	else
	{
		tex2d.Load("rd/enemy-bullet.png");
		spriteRect = sf::IntRect(0, 0, tex2d.sprite.getGlobalBounds().width, tex2d.sprite.getGlobalBounds().height);
	}
}

sf::IntRect Bullet::TileBoundingBox()
{
	return sf::IntRect((int)pos.x, (int)pos.y, 28, 21);
}

void Bullet::setPosition(sf::Vector2f position)
{
	pos = position;
	tex2d.sprite.setPosition(sf::Vector2f(position.x, position.y));
}

sf::Vector2f Bullet::getPosition()
{
	return pos;
}

void Bullet::Update(float deltaTime)
{
	if (reversed)
	{
		pos.y += 150 * deltaTime;
	}
	else
	{
		pos.y -= 150 * deltaTime;
	}

	tex2d.sprite.setPosition(pos);
}

File: libgdx_html_space_invaders/core/src/com/mygdx/game/Bullet.java
package com.mygdx.game;

import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.Input;
import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.badlogic.gdx.math.MathUtils;
import com.badlogic.gdx.math.Rectangle;
import com.badlogic.gdx.math.Vector2;

import java.util.ArrayList;

public class Bullet {

    Texture sprite;
    Vector2 pos;
    Rectangle spriteRect = new Rectangle(0, 0, 6, 36);

    boolean revered = false;

    Bullet(Vector2 pos, boolean revered)
    {
        Create(pos, revered);
    }

    public boolean isRevesed()
    {
        return revered;
    }

    public Rectangle TileBoundingBox()
    {
        return new Rectangle(
                pos.x,
                pos.y,
                spriteRect.width,
                spriteRect.height);
    }

    public void Create(Vector2 _pos, boolean revered)
    {
        this.revered = revered;

        if(revered)
        {
            sprite = new Texture("enemy-bullet.png");
        }
        else
        {
            sprite = new Texture("bullet.png");

        }
        spriteRect = new Rectangle(0, 0, sprite.getWidth(), sprite.getHeight());
        pos = new Vector2(_pos.x - sprite.getDepth()/2 + 10, _pos.y + 20);
    }

    public void Update()
    {
        if(!revered)
            pos.y += 350 * Gdx.graphics.getDeltaTime();
        else
            pos.y -= 350 * Gdx.graphics.getDeltaTime();
    }

    public void Draw(SpriteBatch batch)
    {
        batch.draw(sprite, pos.x, pos.y);
    }

}
